hnn_core/tests/test_viz.py

```diff
@@ -211,4 +222,117 @@ def test_dipole_visualization():
     with pytest.raises(ValueError, match="'beta_dist' must be"):
         net.cell_response.plot_spikes_hist(color={'beta_prox': 'r'})
 
+    # test NetworkPlotter class
```

Comment:
The tests in this module are too long. Generally, each unit test should focus on one specific aspect of the behavior of a function to better isolate location of bugs. That refactor should probably come in a separate PR, but for this one I think the NetworkPlotter should be spun out into its own test at the very least. Ideally it would be broken even further into the main operations you are testing. Here are the separate operations I see from a quick glance: 

1. init  keyword validation 
2. setter validation 
3. simulation validation 
4. movie export 

Creating a pytest fixture for the network will help with redundancies. Also 2nd creating a loop for the pytest raises. You can use the f-strings and dictionary unpacking. Something like:
```python
args = ['xlim', 'ylim', 'zlim']
    for arg in args:
        d_kw = {arg: 'blah'}
        with pytest.raises(TypeError, match=f'{arg} must be'):
            _ = NetworkPlotter(net, **d_kw)
```

Reply:
Thanks for the suggestions @gtdang! Just updated the tests accordingly, let me know what you think